### PR TITLE
Implementation of the resourceUtil mode for the progressive rollout

### PR DESCRIPTION
**Pull request by houshengbo**
Fixed: #160
- The default mode to roll out a new revision is availability, making sure that service availability is the top priory. It will scale up the pods of the new revision first and then scale down the pods of the old revision.
- This PR implemented another mode: resourceUtil to roll out a new revision. This mode makes sure that resource usage is the top priory. It will scale down the pods of the old revision first and then scale up the pods of the new revision.

### CODE REVIEW

File: pkg/reconciler/rolloutorchestrator/strategies/strategy.go [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+					Labels: map[string]string{
+						serving.RevisionLabelKey: revUp.RevisionName,
+						serving.ServiceLabelKey:  ro.Name,
+					},
+					OwnerReferences: []metav1.OwnerReference{
+						*kmeta.NewControllerRef(ro),
+					},
+				},
+				Spec: v1.StagePodAutoscalerSpec{
+					// If the SPA for the new revisions are created, start with 0 for the MinScale, and 1 for MaxScale.
```
Comment (yuzisun) [inline pkg/reconciler/rolloutorchestrator/strategies/strategy.go:139]: The comment is wrong as MaxScale is setting to 1, can we add an explanation why it is set this way?

Comment (houshengbo) [reply pkg/reconciler/rolloutorchestrator/strategies/strategy.go:139]: Yes, I will add the explanations.

**Pull request merged**